webui/react/src/pages/ExperimentList.tsx

```diff
@@ -118,6 +122,38 @@ const ExperimentList: React.FC = () => {
 
   usePolling(fetchExperiments);
 
+  const setLabels = useCallback((id) => {
+    return (labels: string[]) => {
+      patchExperiment({
+        body: {
+          labels: labels.reduce((a, c) => ({ ...a, [c]: true }), {}),
+        },
+        experimentId: id })
+        .then(fetchExperiments);
+    };
+
+  }, [ fetchExperiments ]);
+
+  useEffect(() => {
+    const nameColumn: ColumnType<ExperimentItem> = {
+      dataIndex: 'name',
+      render: function nameRenderer(_, record) {
+        return (
+          <div className={css.nameColumn}>
+            {record.name || ''}
+            <TagList className={css.tagList}
+              setTags={setLabels(record.id)} tags={record.config.labels || []} />
+          </div>
+        );
+      },
+      sorter: (a: ExperimentItem, b: ExperimentItem): number => alphanumericSorter(a.name, b.name),
+      title: 'Name',
+    };
+
+    const existingCol = columns.find(col => col.dataIndex === nameColumn.dataIndex);
+    if (!existingCol) columns.splice(1, 0, nameColumn);
```

Comment:
**nit:** `columns` is not really being reused anywhere else but generally speaking with reactive frameworks immutable approach is good to follow. So rather modifying the existing `columns`, creating a new one with the necessary changes. Maybe use of spread operator + `slice`?

Reply:
https://github.com/determined-ai/determined/pull/960/commits/9980ffc595f724d98c44cf33753bf915244661f2 ?

Reply:
if not this way we'd need to set up a new state (useState) :thinking: